File: test_api.py
import requests

# Test user-chat API
response = requests.get('http://localhost:8001/api/v1/user-chat/user/8')
print('User-chat API Status:', response.status_code)
if response.status_code == 200:
    data = response.json()
    print('Records found:', len(data))
    for record in data:
        print('  Conversation:', record['conversation_id'])
else:
    print('Error:', response.text)

# Test chat-history API
response2 = requests.get('http://localhost:8001/api/v1/chat-history/conversation/bdd47818-c62a-4a92-854d-5a87a34f9985')
print('\nChat-history API Status:', response2.status_code)
if response2.status_code == 200:
    data2 = response2.json()
    print('Messages found:', len(data2))
    for msg in data2:
        print('  Input:', msg.get('input_text', 'None')[:50])
        print('  Output:', msg.get('output_text', 'None')[:50])
else:
    print('Error:', response2.text)

File: frontend/public/test-api.js
// Test script để kiểm tra API calls từ frontend
async function testAPICalls() {
  console.log('=== Testing API calls ===');

  try {
    // Test user-chat API
    console.log('Testing user-chat API...');
    const userChatResponse = await fetch('http://localhost:8001/api/v1/user-chat/user/8');
    console.log('User-chat status:', userChatResponse.status);

    if (userChatResponse.ok) {
      const userChatData = await userChatResponse.json();
      console.log('User-chat data:', userChatData);

      // Test chat-history API for each conversation
      for (const record of userChatData) {
        console.log(`Testing chat-history for conversation: ${record.conversation_id}`);
        const chatResponse = await fetch(`http://localhost:8001/api/v1/chat-history/conversation/${record.conversation_id}`);
        console.log('Chat-history status:', chatResponse.status);

        if (chatResponse.ok) {
          const chatData = await chatResponse.json();
          console.log('Chat data:', chatData);
        } else {
          console.error('Chat-history error:', await chatResponse.text());
        }
      }
    } else {
      console.error('User-chat error:', await userChatResponse.text());
    }

    // Test users API
    console.log('Testing users API...');
    const usersResponse = await fetch('http://localhost:8001/api/v1/users/');
    console.log('Users status:', usersResponse.status);

    if (usersResponse.ok) {
      const usersData = await usersResponse.json();
      console.log('Users data length:', usersData.users?.length || 'No users array');
    } else {
      console.error('Users error:', await usersResponse.text());
    }

  } catch (error) {
    console.error('Test error:', error);
  }
}

// Run test
testAPICalls();
